va/validation-authority_test.go

```diff
@@ -62,25 +68,39 @@ const pathMoved = "301"
 const pathUnsafe = "%"
 const pathUnsafe302 = "302-to-unsafe"
 
+func createValidation(token string, enableTLS bool) string {
+	payload, _ := json.Marshal(map[string]interface{}{
+		"type":  "simpleHttp",
+		"token": token,
+		"tls":   enableTLS,
+	})
+	signer, _ := jose.NewSigner(jose.RS256, &TheKey)
+	obj, _ := signer.Sign(payload, "")
+	return obj.FullSerialize()
+}
+
 func simpleSrv(t *testing.T, token string, stopChan, waitChan chan bool, enableTLS bool) {
 	m := http.NewServeMux()
 
+	defaultToken := token
+	currentToken := defaultToken
+
 	m.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		if strings.HasSuffix(r.URL.Path, path404) {
 			t.Logf("SIMPLESRV: Got a 404 req\n")
 			http.NotFound(w, r)
-		} else if strings.HasSuffix(r.URL.Path, pathWrongToken) {
-			t.Logf("SIMPLESRV: Got a wrongtoken req\n")
-			fmt.Fprintf(w, "wrongtoken")
 		} else if strings.HasSuffix(r.URL.Path, pathMoved) {
 			t.Logf("SIMPLESRV: Got a 301 redirect req\n")
+			if currentToken == defaultToken {
+				currentToken = pathMoved
+			}
 			http.Redirect(w, r, "valid", 301)
 		} else if strings.HasSuffix(r.URL.Path, pathFound) {
 			t.Logf("SIMPLESRV: Got a 302 redirect req\n")
+			if currentToken == defaultToken {
+				currentToken = pathFound
+			}
 			http.Redirect(w, r, pathMoved, 302)
-		} else if strings.HasSuffix(r.URL.Path, pathUnsafe302) {
```

Comment:
Yes.  The path used is now the token, which is issued by the server.  So it can't have unsafe characters.  I'll delete the `pathUnsafe` consts.